app/controllers/builds_controller.rb

```diff
@@ -1,6 +1,13 @@
 class BuildsController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :find_project
+  include CurrentProject
+  include ProjectLevelAuthorization
+
+  before_action do
+    find_project(params[:project_id])
+  end
```

Comment:
Just for everyone's information, I went ahead with the changes suggested and created `CurrentProject#current_project` as follows:
```
def current_project
    @project ||= Project.find_by_param!(params[:project_id]) if params.try(:[], :project_id)
  end
```
And updated all the files where we needed to change from using `@project` to start using this helper_method.